RadialDisplacementAux

bwspenc opened the pull request:
closes #7604

bwspenc committed:
Add auxkernels to compute radial displacement ref #7604

bwspenc committed:
Test RadialDisplacementCylinderAux and RadialDisplacementSphereAux closes #7604

bwspenc commented:
@jiangwen84

dschwen commented on modules/tensor_mechanics/include/auxkernels/RadialDisplacementAux.h:
```diff
@@ -0,0 +1,56 @@
+public:
+
+  /**
+   * Calcualtes the radial displacement for cylindrical or spherical
```
Typo calculates

dschwen commented on modules/tensor_mechanics/src/auxkernels/RadialDisplacementAux.C:
```diff
@@ -0,0 +1,212 @@
+      _coord_system = _subproblem.getCoordSystem(sbd);
+    else if (_subproblem.getCoordSystem(sbd) != _coord_system)
+      mooseError("RadialDisplacementAux requires that all subdomains have the same coordinate type");
+  }
```
This loop should be factored out. I just wrote one like it in the tensor mechanics action.

bwspenc replied:
My new version still has this.  We discussed this, and agreed that we would create a separate issue to address this.

dschwen commented:
I think the design could be simplified by tossing out most parameters and the have origin (just set it to 0 by default and get rid of the extra coffee path)

permcody commented:
On another note - it'd be nice if a BISON person would take a look at the failed test here:
```zrh_formation.zrh_formation```

It exodiffs quite regularly.

dschwen commented on modules/tensor_mechanics/include/auxkernels/RadialDisplacementSphereAux.h:
```diff
@@ -0,0 +1,46 @@
+  /**
+   * Calculates the radial displacement for spherical geometries.
+   * Works for 3D, 2D axisymmetric, and 1D geometries
+   */
```
This comment needs to be right before the class, not the constructor. (same with the other class)

dschwen commented on modules/tensor_mechanics/src/auxkernels/RadialDisplacementCylinderAux.C:
```diff
@@ -0,0 +1,117 @@
+  }
+
+  _ndisp = coupledComponents("displacements");
+  _disp_vals.resize(_ndisp);
```
these two lines should be replaced by initializer list entries.

dschwen commented on modules/tensor_mechanics/src/auxkernels/RadialDisplacementCylinderAux.C:
```diff
@@ -0,0 +1,117 @@
+    if (_ndisp == 2)
+    {
+      rad_vec = current_point - _origin;
+    }
```
no braces here

dschwen commented on modules/tensor_mechanics/src/auxkernels/RadialDisplacementCylinderAux.C:
```diff
@@ -0,0 +1,117 @@
+  Real rad_disp = 0.0;
+  Point current_point (*_current_node);
+
+  if (_coord_system == Moose::COORD_XYZ)
```
usually we use a switch rather than an if else if else

bwspenc commented:
@dschwen I addressed all of your comments, but they're not going away like they used to. Maybe that's a new feature of github -- it's been a few weeks since I've used it.

jiangwen84 commented:
@bwspenc  This works for nodal variable. I think it will be good to check if isNodal() is true in the constructor?

dschwen commented on modules/tensor_mechanics/src/auxkernels/RadialDisplacementSphereAux.C:
```diff
@@ -0,0 +1,87 @@
+  if ((_coord_system == Moose::COORD_XYZ && _ndisp == 3) ||
+      (_coord_system == Moose::COORD_RZ && _ndisp == 2))
+  {
+    Point current_point (*_current_node);
```
I think @jiangwen84 is right. For elemental aux vars this will derefernce a null pointer. Also: space.

bwspenc replied:
Yeah, good point, @jiangwen84.  That would prevent the issue we had when we accidentally used this on an element AuxVariable.  I put a check in the constructor.
